test/driver.js

```diff
@@ -66,6 +66,51 @@ function writeSVG(svgElement, ctx, resolve, reject) {
   };
 }
 
+function inlineImages(images) {
+  const imagePromises = [];
+  for (let i = 0, ii = images.length; i < ii; i++) {
+    const imagePromise = new Promise(function (resolve, reject) {
+      const xhr = new XMLHttpRequest();
+      xhr.responseType = "blob";
+      xhr.onload = function () {
+        const reader = new FileReader();
+        reader.onloadend = function () {
+          resolve(reader.result);
+        };
+        reader.readAsDataURL(xhr.response);
+      };
+      xhr.onerror = function (e) {
+        reject(new Error("Error fetching inline image " + e));
+      };
+      xhr.open("GET", images[i].src);
+      xhr.send();
+    });
+    imagePromises.push(imagePromise);
+  }
+  return imagePromises;
+}
+
+async function resolveImages(node) {
+  const images = node.getElementsByTagName("img");
+  const imagePromises = inlineImages(images);
+
+  await Promise.all(imagePromises).then(function (data) {
```

Comment:
Nit: since you are in an async function, you could avoid the `then' and so reduce the indentation level:
```
await Promise.all(imagePromises);
const loadedPromises = [];
...
```

Reply:
You probably mean `const data = await Promise.all(imagePromises);` here, since otherwise this function would be broken :-)

Reply:
Yep :)